arrow-cast/src/display.rs

```diff
@@ -309,9 +379,10 @@ fn append_map_field_string(
 ///
 /// Note this function is quite inefficient and is unlikely to be
 /// suitable for converting large arrays or record batches.
-pub fn array_value_to_string(
+fn array_value_to_string_internal(
     column: &ArrayRef,
     row: usize,
+    datetime_format_opt: &Option<String>,
```

Comment:
FYI I think the typical rust way to do this (and not require an owned `String`) is:

```suggestion
    datetime_format_opt: Option<&str>,
```

And then at callsites with `val: Option<String>` instead of:

```rust
array_value_to_string_internal(foo, bar, &val)
```

You do :

```rust
array_value_to_string_internal(foo, bar, val.as_ref)
```